src/fetch-wrapper.ts

```diff
@@ -123,7 +126,7 @@ export default function fetchWrapper(
         throw error;
       }
 
-      return getResponseData(response);
+      return getResponseData(requestOptions, response);
```

Comment:
I think instead of passing through requestOptions, I'd add the check for `requestOptions.request?.asStream` before calling `getResponseData`.

I wonder if we should make the option more generic, like `.request.parseResponse` which can be set to false. If it's false, return the `response` object instead of `response.data`

Reply:
If I had to write it for myself, I would have detected applilcation/octet-stream and application/x-gzip and just passed the stream. Maybe it could be useful to actually provide a function, which is used instead of getResponseData?

Idk. 

I would apply the changes as you wish.

Reply:
Yes I was thinking the same, but there might be cases where people prefer to download the whole body as buffer instead of handling a stream.

I'd say let's implement `.request.parseResponse` and try it out.